### PR TITLE
Disjoint pool shared limits

### PR DESCRIPTION
**Pull request by igchor**
Relies on: https://github.com/oneapi-src/unified-memory-framework/pull/28

### CODE REVIEW

**Review by ldorau — CHANGES_REQUESTED** [review]
It does not build on Windows:
```
[  2%] Building CXX object src/CMakeFiles/unified_memory_framework.dir/pool/disjoint/pool_disjoint_impl.cpp.obj
pool_disjoint_impl.cpp
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): error C2672: 'std::max': no matching overloaded function found
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\utility(51): note: could be '_Ty std::max(std::initializer_list<_Elem>)'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: '_Ty std::max(std::initializer_list<_Elem>)': expects 1 arguments - 2 provided
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\utility(48): note: or       '_Ty std::max(std::initializer_list<_Elem>,_Pr)'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: '_Ty std::max(std::initializer_list<_Elem>,_Pr)': could not deduce template argument for 'std::initializer_list<_Elem>' from 'size_t'
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\utility(41): note: or       'const _Ty &std::max(const _Ty &,const _Ty &) noexcept(<expr>)'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: 'const _Ty &std::max(const _Ty &,const _Ty &) noexcept(<expr>)': template parameter '_Ty' is ambiguous
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: could be 'unsigned long'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: or       'size_t'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: 'const _Ty &std::max(const _Ty &,const _Ty &) noexcept(<expr>)': could not deduce template argument for 'const _Ty &' from 'unsigned long'
C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\include\utility(31): note: or       'const _Ty &std::max(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)'
C:\Users\ldorau\work\unified-memory-framework\src\pool\disjoint\pool_disjoint_impl.cpp(296): note: 'const _Ty &std::max(const _Ty &,const _Ty &,_Pr) noexcept(<expr>)': expects 3 arguments - 2 provided
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E cmake_cl_compile_depends --dep-file=CMakeFiles\unified_memory_framework.dir\pool\disjoint\pool_disjoint_impl.cpp.obj.d --working-dir=C:\Users\ldorau\work\unified-memory-framework\build\src --filter-prefix="Note: including file: " -- C:\PROGRA~1\MIB055~1\2022\PROFES~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\cl.exe @C:\Users\ldorau\AppData\Local\Temp\nm38D9.tmp' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\nmake.exe" -s -f src\CMakeFiles\unified_memory_framework.dir\build.make /nologo -SL                 src\CMakeFiles\unified_memory_framework.dir\build' : return code '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files\Microsoft Visual Studio\2022\Professional\VC\Tools\MSVC\14.35.32215\bin\HostX64\x64\nmake.exe" -s -f CMakeFiles\Makefile2 /nologo -S                  all' : return code '0x2'
Stop.
```

**Comment by igchor** [comment]
@ldorau fixed

File: test/c_api/disjoint_pool.c [hunk 1]
```diff
@@ -24,6 +24,28 @@ void test_disjoint_pool_default_params() {
...
+    umf_memory_pool_handle_t pool = NULL;
+    struct umf_disjoint_pool_params params = umfDisjointPoolParamsDefault();
+
+    struct umf_disjoint_pool_shared_limits *limits =
+        umfDisjointPoolSharedLimitsCreate(1024);
+    params.SharedLimits = limits;
+
+    retp = umfPoolCreate(&UMF_DISJOINT_POOL_OPS, provider, &params, &pool);
+
+    // TODO: use asserts
```
Comment (ldorau) [inline test/c_api/disjoint_pool.c:39]: Are you going to address this TODO in this PR?

**Review by ldorau — APPROVED** [review]

File: test/c_api/disjoint_pool.c (hunk 1 above)
Comment (igchor) [reply test/c_api/disjoint_pool.c:39]: Not in this one. I've created a task in Jira to add some helpers for C (asserts, etc.). Once this is done, we can address this.

Comment (ldorau) [reply test/c_api/disjoint_pool.c:39]: OK

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**